components/logins/src/error.rs

```diff
@@ -102,7 +98,170 @@ impl Error {
                 InvalidLogin::NoTarget => "InvalidLogin::NoTarget",
                 InvalidLogin::IllegalFieldValue { .. } => "InvalidLogin::IllegalFieldValue",
             },
-            ErrorKind::ProtobufDecodeError(_) => "BufDecodeError",
+        }
+    }
+}
+
+// TODO: Move the above to somewhere else - everything below is our "public" error list.
+// Probably copy the pattern started by fxa-client - everything above moves to `internal/`?
+// Originally exposed manually via LoginsStorageException.kt.
+
+#[derive(Debug, thiserror::Error)]
+pub enum LoginsError {
```

Comment:
I propose we call it `LoginsRecordError` feels like almost all these issues is with some kind of record issue so makes the most sense for the final error we use.

Reply:
By my count, less than 1/2 of them are actually talking about the record. It might make sense to have 2 error enums - one for invalid record, one for other errors, but even that's gray - eg, is `NoSuchRecord` one of them? Probably not - so there would only be 2 errors in that enum?

Reply:
I agree here. I think splitting the enums would make things a little more confusing for the consumers of which to use and whatnot. I think going by what each of them interface with it seems like majority of them are when there is a LoginsStorage issue so new proposal would be base class of `LoginsStorageError` and `Unexpected` should be `UnexpectedLoginsStorageError` a bit verbose but is quite to the point!